### PR TITLE
feat(home-page): build cms driven homepage content using modular blocks

### PR DESCRIPTION
**Pull request by kibo-geetanshuchhabra**

### CODE REVIEW

File: lib/operations/get-page/index.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import Stack from '../../cms/content-stack'
+
+export const getHomePageCMSRes = async () => {
```
Comment (kibo-kevinwatts) [inline lib/operations/get-page/index.ts:3]: Lets modify these params a little so we can use re-use one getPage operation
and modify the response type so we can add more props later
```
export const getPage = async ({ contentTypeUid, referenceFieldPath}) => {
         const response = await Stack.getEntry({ ...

        return {
               components: response[0][0]?.page_components
        }
}
```

File: lib/operations/get-page/index.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      'page_components.hero_carousel.hero_carousel_items',
+      'page_components.large_promo_blocks.large_promo_blocks',
+      'page_components.small_promo_blocks.small_promo_blocks',
+    ],
+    jsonRtePath: [],
+  })
+  return response[0][0]?.page_components
+}
+
+export const getProductRes = async () => {
```
Comment (kibo-kevinwatts) [inline lib/operations/get-page/index.ts:16]: Can we modify or create something like "product_detail_page" in the CMS? That also has a "page_components" block?

File: pages/index.tsx [hunk 3]
```diff
@@ -1,19 +1,25 @@
...
 
+interface HomePageProps {
+  cmsResults: any
+}
+
 export async function getStaticProps(context: GetStaticPropsContext) {
   const { locale } = context
   const categoriesTree: CategoryCollection = await getCategoryTree()
-
+  const cmsResults = await getHomePageCMSRes()
```
Comment (kibo-kevinwatts) [inline pages/index.tsx:18]: this is a nitpick, can we rename this to `cmsPage`? so it would be `cmsPage?.components?.map`

**Review by kibo-kevinwatts — CHANGES_REQUESTED** [review]
Couple of notes but this is great!

File: lib/operations/get-page/index.ts (hunk 1 above)
Comment (kibo-geetanshuchhabra) [reply lib/operations/get-page/index.ts]: Resolved

File: lib/cms/content-stack/index.js [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+      const query = Stack.ContentType(contentTypeUid).Query()
+      if (referenceFieldPath) query.includeReference(referenceFieldPath)
+      query
+        .includeOwner()
+        .toJSON()
+        .find()
+        .then(
+          (result) => {
+            jsonRtePath &&
+              Utils.jsonToHTML({
```
[the remark this answers is not in the document]
Comment (kibo-geetanshuchhabra) [reply lib/cms/content-stack/index.js]: Resolved

File: components/home/CmsComponent/CmsComponent.tsx [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+  const { content } = props
+  const name = Object.keys(content)[0]
+  const ComponentMapping = props.ComponentMap || DefaultComponentMap
+  const mapping = ComponentMapping[name]
+  const Component = mapping?.component
+  const cmsProps = mapping.mapDataToProps(content)
+  return Component ? <Component {...cmsProps} /> : null
+}
+
+export default CmsComponent
```
[the remark this answers is not in the document]
Comment (kibo-geetanshuchhabra) [reply components/home/CmsComponent/CmsComponent.tsx:131]: Resolved

**Review by kibo-kevinwatts — APPROVED** [review]

**Pull request merged**